devicetypes/smartthings/zigbee-multi-button.src/zigbee-multi-button.groovy

```diff
@@ -70,11 +70,11 @@ def parseAttrMessage(description) {
 		map = getButtonEvent(descMap)
 	} else if(descMap?.clusterInt == 0xFC80) {
 		def buttonNumber
-		buttonNumber = Integer.valueOf(descMap?.command[1].toInteger()) + 1
+		buttonNumber = Integer.valueOf(descMap?.command[1].toInteger())
        
 		log.debug "Number is ${buttonNumber}"
 		def event = createEvent(name: "button", value: "pushed", data: [buttonNumber: buttonNumber], descriptionText: "pushed", isStateChange: true)
-		if (buttonNumber != 1) {
+		if (buttonNumber != 0) {
```

Comment:
I believe the real problem is this check here. On any button press you should create an event on the main device with `buttonNumber` set to the actual button number (1-4) AND you should also generate an event on the appropriate child device. Take a look at how `getButtonResult` does this for an example.

Reply:
button number is 0,1,2,3
when number is 0, event is sent to main button,otherwise sent to child buttons

in getButtonResult, event only sent to child buttons, since this function is implemented for Iris KeyFob and vid of KeyFob is generic-4-button, only 4 child buttons are shown in device plugin, so all the event should be sent to child buttons.

I've also tried to use generic-4-button instead of generic-4-button-alt, and sent event to 4 child buttons like KeyFob, but as I said in slack, the status in device card is main component, doesn't match any of child device. so I have to use generic-4-button-alt and create 1 main button and 3 child buttons .

Reply:
I think this combination of things will work:
* Use `generic-4-button`
* Create 4 components using `addChildDevice`
* On every button event, create an event on the main component with `buttonNumber` set to the appropriate number (1-4) and also create an event on the appropriate child component with `buttonNumber` set to 1